sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -239,5 +246,855 @@ public virtual Response<TimeSeriesModelSettings> UpdateModelSettingsDefaultTypeI
+        /// </remarks>
+        /// <example>
+        /// <code snippet="Snippet:TimeSeriesInsightsSampleGetInstancesByNames">
+        /// </code>
+        /// </example>
+        /// <exception cref="ArgumentNullException">
+        /// The exception is thrown when <paramref name="timeSeriesNames"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// The exception is thrown when <paramref name="timeSeriesNames"/> is empty.
+        /// </exception>
+        public virtual async Task<Response<InstancesOperationResult[]>> GetInstancesAsync(
+            IEnumerable<string> timeSeriesNames,
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNullOrEmpty(timeSeriesNames, nameof(timeSeriesNames));
+
+                InstancesBatchRequest batchRequest = new InstancesBatchRequest()
+                {
+                    Get = new InstancesRequestBatchGetOrDelete()
+                };
+
+                foreach (string timeSeriesName in timeSeriesNames)
+                {
+                    batchRequest.Get.Names.Add(timeSeriesName);
+                }
+
+                Response<InstancesBatchResponse> executeBatchResponse = await _timeSeriesInstancesRestClient
+                    .ExecuteBatchAsync(batchRequest, _clientSessionId, cancellationToken)
+                    .ConfigureAwait(false);
+
+                return Response.FromValue(executeBatchResponse.Value.Get.ToArray(), executeBatchResponse.GetRawResponse());
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets Time Series instances by instance names synchronously.
+        /// </summary>
+        /// <param name="timeSeriesNames">List of names of the Time Series instance to return.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// List of instance or error objects corresponding by position to the array in the request. Instance object is set when operation is successful
+        /// and error object is set when operation is unsuccessful.
+        /// </returns>
+        /// <seealso cref="GetInstancesAsync(IEnumerable{string}, CancellationToken)">
+        /// See the asynchronous version of this method for examples.
+        /// </seealso>
+        /// <exception cref="ArgumentNullException">
+        /// The exception is thrown when <paramref name="timeSeriesNames"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
```

Comment:
same comment as above here and below

Reply:
same response as above here and below :)